var/spack/repos/builtin/packages/py-openpyxl/package.py

```diff
@@ -10,20 +10,17 @@ class PyOpenpyxl(PythonPackage):
     """A Python library to read/write Excel 2010 xlsx/xlsm files"""
 
     homepage = "https://openpyxl.readthedocs.org/"
-    pypi = "openpyxl/openpyxl-3.0.3.tar.gz"
+    pypi = "openpyxl/openpyxl-3.1.2.tar.gz"
 
+    version("3.1.2", sha256="a6f5977418eff3b2d5500d54d9db50c8277a368436f4e4f8ddb1be3422870184")
+    version("3.0.7", sha256="6456a3b472e1ef0facb1129f3c6ef00713cebf62e736cd7a75bcc3247432f251")
     version("3.0.3", sha256="547a9fc6aafcf44abe358b89ed4438d077e9d92e4f182c87e2dc294186dc4b64")
-    version("2.4.5", sha256="78c331e819fb0a63a1339d452ba0b575d1a31f09fdcce793a31bec7e9ef4ef21")
-    version("2.2.0", sha256="c34e3f7e3106dbe6d792f35d9a2f01c08fdd21a6fe582a2f540e39a70e7443c4")
-    version("1.8.6", sha256="aa11a4acd2765392808bca2041f6f9ba17565c72dccc3f5d876bf78effa06126")
-
-    depends_on("python@3.6:", when="@3.0:", type=("build", "run"))
-    depends_on("python@2.7:2.8,3.5:", when="@2.6:", type=("build", "run"))
-    depends_on("python@2.7:2.8,3.4:", when="@2.5:", type=("build", "run"))
-    depends_on("python@2.6:2.8,3.3:", when="@2.1:", type=("build", "run"))
-    depends_on("python@2.6:", type=("build", "run"))
 
+    depends_on("python@3.7:", when="@3.0:", type=("build", "run"))
     depends_on("py-setuptools", type="build")
-
-    depends_on("py-jdcal", when="@2.2:", type=("build", "run"))
     depends_on("py-et-xmlfile", when="@2.4:", type=("build", "run"))
+    depends_on("py-pip@18.0:", when="@3.1.2:", type=("build", "run"))
+    depends_on("py-lxml@4.2.0:", when="@3.1.2:", type=("build", "run"))
+    depends_on("py-pillow", when="@3.1.2:", type=("build", "run"))
+    depends_on("py-tox", when="@3.1.2:", type=("build", "run"))
+    depends_on("py-pandas", when="@3.1.2:", type=("build", "run"))
```

Comment:
These dependencies are not needed. They are not listed in `setup.py` and the packages build fine without them.

Reply:
They are in the requirements.txt

Reply:
They are not installed if you would install the package without spack but with pip because they are not listed in `setup.py`.
And according to [this documentation](https://foss.heptapod.net/openpyxl/openpyxl/-/blob/3.1.2/doc/development.rst) they seem to be only dev dependencies or to support special libraries (like `py-pillow` for images). If you think that (some of them) are needed because you want the functionality, you can maybe add variants (for example variants for the different cases described in [installation tutorial](https://openpyxl.readthedocs.io/en/stable/tutorial.html))